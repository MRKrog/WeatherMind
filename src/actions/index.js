export const setWeather = (weather) => ({
  type: "SET_WEATHER",
  weather
})

export const setCurrent = (info) => ({
  type: "SET_CURRENT",
  info
})

export const setDetails = (details) => ({
  type: "SET_DETAILS",
  details,
})

export const setHourly = (info) => ({
  type: "SET_HOURLY",
  info
})

export const setToday = (info) => ({
  type: "SET_TODAY",
  info
})

export const setWeek = (info) => ({
  type: "SET_WEEK",
  info
})

export const setError = (error) => ({
  type: "SET_ERROR",
  error
})

export const setLoading = (loading) => ({
  type: "SET_LOADING",
  loading
})
