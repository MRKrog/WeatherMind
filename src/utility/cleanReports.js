import moment from "moment";

export const cleanCurrently = (data) => {
  const { currently } = data
  return {
    time: currently.time,
    icon: currently.icon,
    summary: currently.summary,
    apparentTemperature: Math.round(currently.apparentTemperature),
    temperature: Math.round(currently.temperature),
    visibility: currently.visibility,
    humidity: currently.humidity,
    windSpeed: currently.windSpeed,
    pressure: currently.pressure,
    uvIndex: currently.uvIndex
  }
}

export const cleanHourly = (report) => {
  const hourlyArray = report.data.reduce((acc, hour, idx) => {
    if(idx < 24) {
      let info = {
        time: moment.unix(hour.time).format("HH"),
        icon: hour.icon,
        temperature: Math.round(hour.temperature),
      }
      acc.push(info)
    }
    return acc
  },[])
  return hourlyArray;
}

export const cleanToday = (report) => {
  const today = report.data[0];
  return {
    time: moment.unix(today.time).format("dddd"),
    tempHigh: Math.round(today.temperatureHigh),
    tempLow: Math.round(today.temperatureLow),
    summary: today.summary,
    sunriseTime: moment.unix(today.sunriseTime).format("LT"),
    sunsetTime:  moment.unix(today.sunsetTime).format("LT"),
    windSpeed: today.windSpeed,
    humidity: today.humidity * 100,
    visibility: today.visibility,
    uvIndex: today.uvIndex,
    precipProbability: today.precipProbability * 100,
    moonPhase: today.moonPhase * 100,
  }
}

export const cleanWeek = (report) => {
  const weeklyArray = report.data.reduce((acc, day, idx) => {
    if(idx > 0) {
      let info = {
        time: moment.unix(day.time).format("dddd"),
        tempHigh: Math.round(day.temperatureHigh),
        tempLow: Math.round(day.temperatureLow),
        icon: day.icon,
      }
      acc.push(info)
    }
    return acc
  },[])
  return weeklyArray;
}
