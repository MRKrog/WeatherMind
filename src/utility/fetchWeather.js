
export const fetchWeather = async (url) => {
  try {
    const response = await fetch(url)
    if(!response.ok) { throw new Error('Bad Call')}
    const data = await response.json()
  } catch (error) {
    console.log(error.message);
  }
}
