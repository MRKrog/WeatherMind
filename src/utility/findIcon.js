import clearNight from '../media/icons/clear-night.svg';
import clearDay from '../media/icons/clear-day.svg';
import partlyCloudyDay from '../media/icons/partly-cloudy-day.svg';
import partlyCloudyNight from '../media/icons/partly-cloudy-night.svg';
import cloudy from '../media/icons/cloudy.svg';
import rain from '../media/icons/rain.svg';
import sleet from '../media/icons/sleet.svg';
import snow from '../media/icons/snow.svg';
import wind from '../media/icons/wind.svg';
import fog from '../media/icons/fog.svg';

export const findIcon = (icon) => {
  switch (icon) {
    case 'clear-day':
      return clearDay
    case 'clear-night':
      return clearNight
    case 'partly-cloudy-day':
      return partlyCloudyDay
    case 'partly-cloudy-night':
      return partlyCloudyNight
    case 'cloudy':
      return cloudy
    case 'rain':
      return rain
    case 'sleet':
      return sleet
    case 'snow':
      return snow
    case 'wind':
      return wind
    case 'fog':
      return fog
    default:
      return cloudy
  }
}
