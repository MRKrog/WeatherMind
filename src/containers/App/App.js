import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Route, NavLink } from 'react-router-dom';

import { Header } from '../Header/Header';
import { Search } from '../Search/Search';
import CurrentDisplay from '../CurrentDisplay/CurrentDisplay';
import HourlyDisplay from '../HourlyDisplay/HourlyDisplay';
import WeeklyDisplay from '../WeeklyDisplay/WeeklyDisplay';
import WeatherDetails from '../WeatherDetails/WeatherDetails'
import { Loading } from '../../components/Loading/Loading';

import { setCurrent, setDetails, setHourly, setToday, setWeek } from '../../actions/index';
import { cleanCurrently, cleanHourly, cleanToday, cleanWeek } from '../../utility/cleanReports';

export class App extends Component {
  constructor() {
    super()
    this.state = {
      loading: true,
    }
  }

  componentDidMount() {
    this.getUserLocation();
  }

  getWeather = async (latitude, longitude) => {
    console.log('in get', latitude);
    console.log('in get', longitude);
    try {
      const response = await fetch(`http://localhost:3001/api/v1/weather/${latitude}/${longitude}`)
      const data = await response.json()

      console.log(data);
      const currentData = cleanCurrently(data[0])
      this.props.setCurrent(currentData)
      const hourlyData = cleanHourly(data[0].hourly)
      this.props.setHourly(hourlyData)
      const todayData = cleanToday(data[0].daily)
      this.props.setToday(todayData)
      const weekData = cleanWeek(data[0].daily)
      this.props.setWeekly(weekData)
      this.props.setDetails(data[1])
      this.setState({ loading: false })
    } catch (error) {
      console.log(error.message);
    }
  }

  getUserLocation = () => {
    navigator.geolocation.getCurrentPosition(position => {
      let latitude = position.coords.latitude
      let longitude = position.coords.longitude
      this.getWeather(latitude, longitude)
    });
  }

  handleSearch = (latitude, longitude) => {
    this.getWeather(latitude,longitude)
  }

  render() {
    const { loading } = this.state;
    return (
      <div className="App">
        <div className="marvel-device iphone-x">
          <div className="notch">
              <div className="camera"></div>
              <div className="speaker"></div>
          </div>
          <div className="top-bar"></div>
          <div className="sleep"></div>
          <div className="bottom-bar"></div>
          <div className="volume"></div>
          <div className="overflow">
              <div className="shadow shadow--tr"></div>
              <div className="shadow shadow--tl"></div>
              <div className="shadow shadow--br"></div>
              <div className="shadow shadow--bl"></div>
          </div>
          <div className="inner-shadow"></div>
          <div className="screen">
          {
            loading ?
            <Loading /> :
            <div className="Weather-Content">
              <Route path="/" component={Header} />
              <Route path="/Search" render={() => {
                return <Search handleSearch={this.handleSearch} />
              }}/>
              <CurrentDisplay />
              <HourlyDisplay />
              <WeeklyDisplay />
              <NavLink className="Button-Container" to="/WeatherDetails">
                <div className="Details-Button">
                  <i className="fas fa-plus-square"></i>
                </div>
              </NavLink>
              <Route exact path="/WeatherDetails" component={WeatherDetails} />
            </div>
          }
          </div>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setToday: (data) => dispatch(setToday(data)),
  setCurrent: (data) => dispatch(setCurrent(data)),
  setDetails: (data) => dispatch(setDetails(data)),
  setHourly: (data) => dispatch(setHourly(data)),
  setWeekly: (data) => dispatch(setWeek(data)),
})

App.propTypes = {
  setToday: PropTypes.func,
  setCurrent: PropTypes.func,
  setDetails: PropTypes.func,
  setHourly: PropTypes.func,
  setWeekly: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(App);
