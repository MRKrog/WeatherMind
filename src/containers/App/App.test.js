import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps } from './App';
import { setCurrent, setDetails, setHourly, setToday, setWeek } from '../../actions/index';
import { cleanCurrently, cleanHourly, cleanToday, cleanWeek } from '../../utility/cleanReports';

const mockGeolocation = {
  getCurrentPosition: jest.fn()
    .mockImplementationOnce((success) => Promise.resolve(success({
      coords: {
        latitude: 51.1,
        longitude: 45.3
      }
    })))
};
global.navigator.geolocation = mockGeolocation;


describe('App', () => {

  describe('App Component', () => {
    let wrapper;
    let mockSetCurrent = jest.fn()
    let mockSetDetails = jest.fn()
    let mockSetHourly = jest.fn()
    let mockSetToday = jest.fn()
    let mockSetWeekly = jest.fn()

    beforeEach(() => {
      wrapper = shallow(
        <App setCurrent={mockSetCurrent}
             setDetails={mockSetDetails}
             setHourly={mockSetHourly}
             setToday={mockSetToday}
             setWeekly={mockSetWeekly}
        />
      )
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it('should have a default state', () => {
      expect(wrapper.state()).toEqual({
        loading: true,
      })
    })

    it('should invoke the getUserLocation method when componentDidMount fires', () => {
      const instance = wrapper.instance()
      const spy = spyOn(instance, 'getUserLocation');
      wrapper.instance().componentDidMount()
      expect(spy).toHaveBeenCalled()
    });

    it('should', () => {})

    it('should', () => {})

  })

  describe('mapDispatchToProps', () => {

    it('should call dispatch for setToday', () => {
      const mockData = 'Info'
      const mockDispatch = jest.fn();
      const actionToDispatch = setToday(mockData)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setToday(mockData)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call dispatch for setCurrent', () => {
      //Setup
      const mockData = "current weather";
      const mockDispatch = jest.fn();
      const actionToDispatch = setCurrent(mockData)
      //Execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setCurrent(mockData);
      //Expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch for setDetails', () => {
      //Setup
      const mockData = "details weather";
      const mockDispatch = jest.fn();
      const actionToDispatch = setDetails(mockData);
      //Execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setDetails(mockData);
      //Expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch for setHourly', () => {
      const mockData = 'hourly weather';
      const mockDispatch = jest.fn()
      const actionToDispatch = setHourly(mockData)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setHourly(mockData)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch for setWeekly', () => {
      const mockData = 'weekly weather'
      const mockDispatch = jest.fn();
      const actionToDispatch = setWeek(mockData); //action

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setWeekly(mockData); // prop action

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  })

});
