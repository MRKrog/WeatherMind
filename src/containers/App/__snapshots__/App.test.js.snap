// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="marvel-device iphone-x"
  >
    <div
      className="notch"
    >
      <div
        className="camera"
      />
      <div
        className="speaker"
      />
    </div>
    <div
      className="top-bar"
    />
    <div
      className="sleep"
    />
    <div
      className="bottom-bar"
    />
    <div
      className="volume"
    />
    <div
      className="overflow"
    >
      <div
        className="shadow shadow--tr"
      />
      <div
        className="shadow shadow--tl"
      />
      <div
        className="shadow shadow--br"
      />
      <div
        className="shadow shadow--bl"
      />
    </div>
    <div
      className="inner-shadow"
    />
    <div
      className="screen"
    >
      <Loading />
    </div>
  </div>
</div>
`;
